change: increase build performance by remapping once

marcus8448 opened the pull request:
Instead of remapping both the api jar and the mod jar (which includes all of the api classes anyway), we can just remap the mod jar, then extract the api classes out of it

This is slightly hacky, but avoids doing double the work

jellysquid3 commented:
This is a pretty neat solution. My only question is whether we can still use this API JAR in a Maven publication while attaching the correct metadata. Do you happen to know?

marcus8448 committed:
change: remap complete mod jar, then split api jar out of it
this is slightly hacky, but avoids doing double the work
force pushing this as the diff got messed up

Signed-off-by: marcus8448 <marcus8448_git@outlook.com>

marcus8448 committed:
change: use a dummy fabric.mod.json instead of remap metadata

Signed-off-by: marcus8448 <marcus8448_git@outlook.com>

marcus8448 commented:
Okay, now the api jar includes a basic `fabric.mod.json` (instead of using `Fabric-Loom-Remap: true`) so loom versions <1.1 can use the resulting api jar. Mods JiJ-ing sodium api should no longer be an issue as making `sodium` provide `sodium-api` should prevent fabric from loading the other jars.

> whether we can still use this API JAR in a Maven publication while attaching the correct metadata

The following snippet publishes the remapped api jar which can be used like any other published mod jar
```gradle
plugins {
    id 'maven-publish'
}

publishing {
    publications {
        mavenJava(MavenPublication) {
            artifact(apiJar) {
                builtBy apiJar
            }
        }
    }
    repositories {
    }
}
```
